### ISSUE TITLE
Problem in Accuracy Rate while fine tuning VGG-19

### ISSUE THREAD

**Issue by rizwanasif** [vlfeat/matconvnet#1038]
Dear Experts,

I am trying to finetune VGG19 on my own binary dataset. When i run the code, it trains and gives me error of 0.02 for validation but when i classify each image in validation set myself, it gives the accuracy of 92% more or less.. Am i doing something wrong here? a part of code is attached for reference.

function [net, info] = finetune_vgg(imdb, expDir,varargin)

opts.train.batchSize = 64;
opts.train.numEpochs = 25;
opts.train.continue = false ;  
opts.train.gpus = [1] ;
opts.train.learningRate = [1e-4];
opts.train.weightDecay = 1e-4;
opts.train.momentum = 0.9;
opts.train.expDir = expDir;
opts.train.numSubBatches = 1;
opts.contrastNormalization = true;
opts.lite = false ;
% getBatch options
bopts.useGpu = numel(opts.train.gpus) >  0 ;

if(numel(varargin) > 0)
    netPre = varargin{1};
end

VGGpath = fullfile('./imagenet-vgg-verydeep-19.mat');
pre = load(VGGpath);
% classes = pre.meta.classes.description;

net = dagnn.DagNN.fromSimpleNN(pre,'canonicalNames', true);

net.meta.classes.description = imdb.meta.classes ;
net.meta.classes.name = imdb.meta.classes ;

layerName  = 'fc8';
layerIndex = net.getLayerIndex(layerName)
paramNames = [net.layers(layerIndex).params]
paramIdx = net.getParamIndex(paramNames)
net.layers(layerIndex).block.size = [1 1 4096 2];
net.params(paramIdx(1)).value = single(randn(1,1,4096,2));
net.params(paramIdx(2)).value = single(randn(2,1));
net.layers(43) = [];
net.addLayer('prob_egg', dagnn.SoftMax(), {'prediction'}, {'prob'}, {});
net.addLayer('objective', dagnn.Loss('loss', 'log'), {'prob', 'label'}, {'objective'}, {});
net.addLayer('error', dagnn.Loss('loss', 'classerror'), {'prob','label'}, 'error') ;

% initialization of the weights (CRITICAL!!!!)
if(numel(varargin) > 0)
    initNet_FineTuning(net, netPre);
else
    initNet_xavier(net);
%     initNet(net, 1/100);
end

% do the training!
opts.train.train = find(imdb.images.set == 1);
opts.train.val = find(imdb.images.set == 2);
[net,info] = cnn_train_dag(net, imdb, @(i,b) getBatchDisk(bopts,i,b), opts.train, 'val', find(imdb.images.set == 2)) ;

end

and i run using 
net = finetune_vgg(imdb,'./path',netprevgg);

the output graphs are something like this
[net-train.pdf](https://github.com/vlfeat/matconvnet/files/1241772/net-train.pdf)
